react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+export default class StopWatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: false,
+      count: 0
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleTimer = this.handleTimer.bind(this);
+    this.handleReset = this.handleReset.bind(this);
+    this.start = this.start.bind(this);
+    this.stop = this.stop.bind(this);
+  }
+
+  handleClick() {
+    if (this.state.isClicked === false) {
+      this.setState({ isClicked: true });
+    } else this.setState({ isClicked: false });
+  }
+
+  handleTimer() {
+    if (this.state.isClicked === true) {
+      this.setState({ count: this.state.count + 1 });
+    }
+  }
+
+  handleReset() {
+    if (this.state.isClicked === false) {
+      this.setState({ count: 0 });
+    }
+  }
+
+  start() {
+    this.intervalID = setInterval(this.handleTimer, 1000);
+  }
+
+  // Function to stop setInterval call
+  stop() {
+    clearInterval(this.intervalID);
+  }
+
+  render() {
+    let icon;
+    if (this.state.isClicked === true) {
+      this.stop();
+      this.start();
```

Comment:
Not the best to call these here, this will cause unnecessary work and re-renders, it would be better to start and stop the timer in the `handleClick` method